src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt

```diff
@@ -0,0 +1,128 @@
+
+  private val refreshButton =
+      object :
+          ToolbarDecorator.ElementActionButton(
+              CodyBundle.getString("context-panel.reindex-button-name"), AllIcons.Actions.Refresh) {
+        override fun actionPerformed(p0: AnActionEvent) {
+          CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+            ProgressManager.getInstance()
+                .run(
+                    object :
+                        Task.Backgroundable(
+                            project, CodyBundle.getString("context-panel.in-progress"), false) {
+                      override fun run(indicator: ProgressIndicator) {
+                        try {
+                          isReindexingInProgress.set(true)
+                          val cmd = CommandExecuteParams("cody.search.index-update", emptyList())
+                          agent.server.commandExecute(cmd).get()
+                        } catch (e: Exception) {
+                          val errMsg = e.message ?: e.toString()
+                          Messages.showErrorDialog(
+                              CodyBundle.getString("context-panel.error-message").fmt(errMsg),
+                              CodyBundle.getString("context-panel.error-title"))
+                        } finally {
+                          indicator.stop()
+                          isReindexingInProgress.set(false)
+                        }
+                      }
+                    })
+          }
+        }
+
+        override fun isEnabled(): Boolean {
+          return !isReindexingInProgress.get()
+        }
+
+        override fun isDumbAware(): Boolean {
+          return true
+        }
+
+        private val isReindexingInProgress = AtomicBoolean(false)
+      }
+
+  private val helpButton =
+      object :
+          ToolbarDecorator.ElementActionButton(
+              CodyBundle.getString("context-panel.help-button-name"), AllIcons.Actions.Help) {
+        override fun actionPerformed(p0: AnActionEvent) {
+          BrowserUtil.open("https://docs.sourcegraph.com/cody/core-concepts/keyword-search")
+        }
+
+        override fun isDumbAware(): Boolean {
+          return true
+        }
+      }
+
+  private val tree = run {
+    val treeRoot = CheckedTreeNode(CodyBundle.getString("context-panel.tree-root"))
+    val localProject = CheckedTreeNode(CodyBundle.getString("context-panel.tree-local-project"))
+    val currentRepo =
+        object : CheckedTreeNode(project) {
+          override fun isChecked(): Boolean {
+            return isEnhancedContextEnabled.get()
+          }
+
+          override fun setChecked(checked: Boolean) {
+            isEnhancedContextEnabled.set(checked)
+            CodyAgentService.getInstance(project).restartAgent(project)
+          }
+        }
+    localProject.add(currentRepo)
+    treeRoot.add(localProject)
+    CheckboxTree(ContextRepositoriesCheckboxRenderer(), treeRoot)
+  }
+
+  private val toolbarPanel = run {
```

Comment:
Why these variables (toolbarPanel, tree) are initialized within `run` block? Can we use simple, plain functions here? Like `private fun createToolbarPanel() {...}`).

Reply:
We could. Run block is basically static initializer.
So e.g. in scala it would look like that:

```
val x = {
   val a = 1
   val b = 2
   a + b
}
// x was initialised to 3
```

Reply:
> I kind of intentionally make them initialised statically to show their creation is not context dependent or time sensitive.

Good point!